comp/core/tagger/taggerimpl/collectors/workloadmeta_test.go

```diff
@@ -519,9 +643,10 @@ func TestHandleKubePod(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			collector := &WorkloadMetaCollector{
-				store:      store,
-				children:   make(map[string]map[string]struct{}),
-				staticTags: tt.staticTags,
+				store:                             store,
+				children:                          make(map[string]map[string]struct{}),
+				staticTags:                        tt.staticTags,
+				collectPersistentVolumeClaimsTags: tt.collectPvcTags,
```

Comment:
I find this test brittle. Since we are create the `WorkloadMetaCollector` instance directly and not using `NewWorkloadMetaCollector ` function if tomorrow I change the code inside the function to always set `collectPersistentVolumeClaimsTags` to false the test would still pass but the behaviour in production would be wrong. 

My suggestion is that we use the `NewWorkloadMetaCollector` function to ensure the code in the test are the same as we use in the production.

Reply:
I agree on this 👍
We should do this for all the tests cases, this would introduce some unwanted changes in this PR. As this is for a support case, I can open a card in our board to have it done in the future.

Reply:
Done in https://github.com/DataDog/datadog-agent/pull/26605